fine-tuning.md

```diff
@@ -19,19 +19,79 @@ Some ideas:
+
+**Use cases**:
+`NOTE`: We can even extend the process of transfer learning by unfreezing some layers of pre-trained model and retraining them along with our smaller model. This additional step helps the model to adapt on newer domain-specific task or out of distribution tasks.
+
+- Limited data: when domain-specific dataset size is small, a large model cannot be trained end-to-end without overfitting. However if the model is mostly a frozen general feature extractor, then the subsequent trainable layers are less likely to overfit.
+- Limited compute and time: retraining a large model from scratch requires a lot of compute resources and time. This is unnecessary if similar performance can be achieved through transfer learning (training just part of a large model).
+
+### Fine-tuning
+
+The key difference here is none (or few) of the pre-trained model's weights are frozen. The pre-training process can be considered an intelligent weight initialisation prior to training on a domain-specific dataset. Essentially, the pre-training will leave the model weights close to a global (general) optimum, while the domain-specific training will find a local (task-specific) optimum.
+**Why and when to use Transfer learning?**
+
+Transfer learning is very much useful when we have the following constrains
+
+1. Limited data: Transfer learning is a useful solution when our dataset size i small. There we can leverage the knowledge from pre-trained model and use that (extracted feature) to fit on our smaller task specific dataset.
+2. Training efficiency: Transfer learning is very useful when we are constrained with compute resources. Retraining the model from scratch can be very resource intensive. However the same performance of the model can be achieved through transfer learning without using much compute resource. Hence the training time is also very small compared to retraining the model.
+
+
+### Fine-Tuning
+
+From [Wikipedia’s](https://en.wikipedia.org/wiki/Fine-tuning_(deep_learning)) definition, Fine-tuning is an approach to transfer learning in which weights of a pre-trained model is trained on a new data.  In some case we retrain the whole model on our domain-specific dataset or in other cases, we just fine-tune on only a subset of the layers. Through fine-tuning, we are adapting our existing pre-trained model on a task-specific dataset.
+
+```{figure-md} fine-tuning-architecture
+:class: caption
+![](https://static.premai.io/book/fine-tuning.png)
+
+Fine Tuning
+```
```

Comment:
@Anindyadeep why there are two sections with the same name? In general keep the same structure for both transfer learning and fine-tuning subsections

Reply:
Yes actually both had same structure, but there was some redundancy which I removed, hopefully it looks all okay now.